UEBPrep/app/models/user.rb

```diff
@@ -30,8 +34,15 @@ def generate_api_key
 
   def self.from_omniauth(auth)
     where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
-      user.email = auth.info.email
-      user.password = Devise.friendly_token[0,20]
+      if !user.email
```

Comment:
We shouldn't have to create an email here, we need to just update devise to not use email as a validation